pyqir-generator/tests/test_external_functions.py

```diff
@@ -125,61 +168,77 @@ def test_call_numbers(self) -> None:
         )
 
     def test_wrong_type_conversion(self) -> None:
-        cases: List[Tuple[List[Type], Callable[[SimpleModule], List[Any]]]] = [
-            ([types.BOOL], lambda _: ["true"]),
-            ([types.Int(64)], lambda _: [1.23]),
-            ([types.Int(64)], lambda _: ["123"]),
-            ([types.DOUBLE], lambda _: ["1.23"]),
+        cases: List[Tuple[Callable[[TypeFactory], List[Type]], List[Any]]] = [
+            (lambda types: [types.bool], ["true"]),
+            (lambda types: [types.int(64)], [1.23]),
+            (lambda types: [types.int(64)], ["123"]),
+            (lambda types: [types.double], ["1.23"]),
         ]
 
-        for param_types, get_args in cases:
+        for get_param_types, args in cases:
             mod = SimpleModule("test", 1, 1)
-            args = get_args(mod)
+            types = mod.types
 
             with self.subTest(repr(args)):
                 f = mod.add_external_function(
-                    "test_function", types.Function(param_types, types.VOID)
+                    "test_function",
+                    types.function(types.void, get_param_types(types)),
                 )
 
                 with self.assertRaises(TypeError):
                     mod.builder.call(f, args)
 
     def test_overflow_bool_value(self) -> None:
-        with self.assertRaises(OverflowError):
-            const(types.BOOL, 123)
+        mod = SimpleModule("test", 0, 0)
+        types = mod.types
+        f = mod.add_external_function("f", types.function(types.void, [types.bool]))
+        b = const(mod.types.bool, 123)
+        mod.builder.call(f, [b])
+        self.assertIn("call void @f(i1 true)", mod.ir())
 
-    def test_overflow_int_value(self) -> None:
-        with self.assertRaises(OverflowError):
-            const(types.Int(64), 2**64)
+    def test_overflow_bool_literal(self) -> None:
+        mod = SimpleModule("test", 0, 0)
+        types = mod.types
+        f = mod.add_external_function("f", types.function(types.void, [types.bool]))
+        mod.builder.call(f, [123])
+        self.assertIn("call void @f(i1 true)", mod.ir())
 
-    def test_overflow_conversion(self) -> None:
-        for value in [123, 2**64]:
-            with self.subTest(repr(value)):
-                mod = SimpleModule("test", 0, 0)
-                f = mod.add_external_function(
-                    "test_function", types.Function([types.BOOL], types.VOID)
-                )
+    def test_overflow_int_value(self) -> None:
+        mod = SimpleModule("test", 0, 0)
+        types = mod.types
+        f = mod.add_external_function("f", types.function(types.void, [types.int(32)]))
+        i = const(mod.types.int(32), 2**32 + 123)
+        mod.builder.call(f, [i])
+        self.assertIn("call void @f(i32 123)", mod.ir())
```

Comment:
Does it make sense to add test with negative values?

Reply:
Sure, added. It turns out that negative values are not supported at all. Probably because Inkwell `const_int` takes a u64 value and the PyO3 conversion to unsigned fails if negative.